packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1489 @@
+///    <property1>
+///    <property2>
+///    ...
+///    <propertyN>
+/// ```
+///
+/// See also:
+///
+///  * [DiagnosticsTreeStyle.sparse]
+final TextTreeConfiguration sparseTextConfiguration = new TextTreeConfiguration(
+  prefixLineOne:            '├─',
+  prefixOtherLines:         ' ',
+  prefixLastChildLineOne:   '└─',
+  linkCharacter:            '│',
+  propertyPrefixIfChildren: '│ ',
+  propertyPrefixNoChildren: '  ',
+  prefixOtherLinesRootNode: ' ',
+);
+
+/// Identical to [sparseTextConfiguration] except that the lines connecting
+/// parent to children are dashed.
+///
+/// Example:
+/// ```
+/// <root_name>: <root_description>
+///  │ <property1>
+///  │ <property2>
+///  │ ...
+///  │ <propertyN>
+///  ├─<normal_child_name>: <child_description>
+///  ╎ │ <property1>
+///  ╎ │ <property2>
+///  ╎ │ ...
+///  ╎ │ <propertyN>
+///  ╎ │
+///  ╎ └─<child_name>: <child_description>
+///  ╎     <property1>
+///  ╎     <property2>
+///  ╎     ...
+///  ╎     <propertyN>
+///  ╎
+///  ╎╌<dashed_child_name>: <child_description>
+///  ╎ │ <property1>
+///  ╎ │ <property2>
+///  ╎ │ ...
+///  ╎ │ <propertyN>
+///  ╎ │
+///  ╎ └─<child_name>: <child_description>
+///  ╎     <property1>
+///  ╎     <property2>
+///  ╎     ...
+///  ╎     <propertyN>
+///  ╎
+///  └╌<dashed_child_name>: <child_description>'
+///    <property1>
+///    <property2>
+///    ...
+///    <propertyN>
+/// ```
+///
+/// See also:
+///
+///  * [DiagnosticsTreeStyle.offstage], uses this style for ascii art display.
+final TextTreeConfiguration dashedTextConfiguration = new TextTreeConfiguration(
+  prefixLineOne:            '╎╌',
+  prefixLastChildLineOne:   '└╌',
+  prefixOtherLines:         ' ',
+  linkCharacter:            '╎',
+  // Intentionally not set as a dashed line as that would make the properties
+  // look like they were disabled.
+  propertyPrefixIfChildren: '│ ',
+  propertyPrefixNoChildren: '  ',
+  prefixOtherLinesRootNode: ' ',
+);
+
+/// Dense text tree configuration that minimizes horizontal whitespace.
+///
+/// Example:
+/// ```
+/// <root_name>: <root_description>
+/// │ <property1>
+/// │ <property2>
+/// │ ...
+/// │ <propertyN>
+/// │
+/// ├<child_name>: <child_description>
+/// │ <property1>
+/// │ <property2>
+/// │ ...
+/// │ <propertyN>
+/// │
+/// └<child_name>: <child_description>'
+///   <property1>
+///   <property2>
+///   ...
+///   <propertyN>
+/// ```
+///
+/// See also:
+///
+///  * [DiagnosticsTreeStyle.dense]
+final TextTreeConfiguration denseTextConfiguration = new TextTreeConfiguration(
+  prefixLineOne:            '├',
+  prefixOtherLines:         '',
+  prefixLastChildLineOne:   '└',
+  linkCharacter:            '│',
+  propertyPrefixIfChildren: '│',
+  propertyPrefixNoChildren: ' ',
+  prefixOtherLinesRootNode: '',
+  addBlankLineIfNoChildren: false,
+);
+
+/// Configuration that draws boxes around nodes.
+///
+/// Typically this style should be used to for leaf nodes such as [TextSpan] to
```

Comment:
One minor concern is the name leaf makes it seem like this style should be used any time you have a leaf when it should really only be used when you want to draw a box around a leaf node.
something like boxLeaf, outlinedLeaf, or emphasizeLeaf might be clearer but none of those names is much better.